### PR TITLE
Add Test_GetChainFeePriceUpdates

### PR DESCRIPTION
**Pull request by asoliman92**

### CODE REVIEW

File: contracts/.changeset/odd-maps-fry.md
Comment (makramkd) [inline contracts/.changeset/odd-maps-fry.md:1]: nit: not needed.

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 1]
```diff
@@ -2,6 +2,10 @@ package ccipreader
 
 import (
 	"context"
+
+	evmconfig "github.com/smartcontractkit/chainlink/v2/core/capabilities/ccip/configs/evm"
+
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: goimports

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 2]
```diff
@@ -47,6 +51,10 @@ const (
 	chainD  = cciptypes.ChainSelector(4)
 )
 
+var (
+	defaultGasPrice = big.NewInt(1e9)
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: Use `assets.GWei` instead

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 3]
```diff
@@ -479,14 +487,86 @@ func TestCCIPReader_Nonces(t *testing.T) {
...
+			LinkToken:                    linkAddress,
+			TokenPriceStalenessThreshold: uint32(24 * 60 * 60),
+		},
+		[]common.Address{auth.From},
+		[]common.Address{wethAddress, linkAddress},
+		[]fee_quoter.FeeQuoterTokenPriceFeedUpdate{},
+		[]fee_quoter.FeeQuoterTokenTransferFeeConfigArgs{},
+		[]fee_quoter.FeeQuoterPremiumMultiplierWeiPerEthArgs{
+			{
+				PremiumMultiplierWeiPerEth: 9e17, // 0.9 ETH
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: I think this means 90% and not 0.9ETH cc @0xnogo

Comment (makramkd) [reply core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: i.e a 10% discount

Comment (0xnogo) [reply core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: @makramkd I think you are referring to the `gasMultiplierWeiPerEth`. But here it's the PremiumMultiplierWeiPerEth

Comment (0xnogo) [reply core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: I think this one is set to pay 10% if you are paying with Link token. I think it needs to have 1e18 unit, so what we have here is correct:

```
    // Calculate number of fee tokens to charge.
    // Total USD fee is in 36 decimals, feeTokenPrice is in 18 decimals USD for 1e18 smallest token denominations.
    // Result of the division is the number of smallest token denominations.
    return ((premiumFee * s_premiumMultiplierWeiPerEth[message.feeToken]) + executionCost + dataAvailabilityCost)
      / feeTokenPrice;
```

executionCost + dataAvailabilityCost => e36
premium (e18) * s_premiumMultiplierWeiPerEth[message.feeToken] (e18) => e36

File: core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go [hunk 4]
```diff
@@ -2,6 +2,7 @@ package ccipreader
 
 import (
 	"context"
+
```
Comment (makramkd) [inline core/capabilities/ccip/ccip_integration_tests/ccipreader/ccipreader_test.go]: Get rid of newline

**Review by makramkd — APPROVED** [review]

**Pull request merged**